receiver/dockerstatsreceiver/docker.go

```diff
@@ -194,6 +194,61 @@ func (dc *dockerClient) toStatsJSON(
 	return &statsJSON, nil
 }
 
+func (dc *dockerClient) ContainerEventLoop(ctx context.Context) {
```

Comment:
The context you use is incorrect, because it is derived from the `Start` call which may be canceled as part of ensuring start of the components happens with a deadline as an example.

So you need to use either a separate channel for cancelation or derive the context from background.

Reply:
No. Start context can have associated a deadline to ensure the collector starts in x seconds (and by doing that it is always recommend to cancel the context even if the request finish successfully). So that will cancel the loop which is not expected. I will add a comment to the Start interface that context cannot be saved or used in any async work

Reply:
Thanks, it's unclear to me how that context should or shouldn't be used otherwise.

Reply:
Trying to clarify this: https://github.com/open-telemetry/opentelemetry-collector/pull/1754